libcodechecker/libhandlers/cmd.py

```diff
@@ -60,27 +62,45 @@ def get_argparser_ctor_args():
     }
 
 
-def __add_common_arguments(parser, has_matrix_output=False):
+def __add_common_arguments(parser,
+                           needs_product_url=True,
+                           has_matrix_output=False):
     """
     Add some common arguments, like server address and verbosity, to parser.
     """
 
     common_group = parser.add_argument_group('common arguments')
 
-    common_group.add_argument('--host',
-                              type=str,
-                              dest="host",
-                              default="localhost",
-                              required=False,
-                              help="The address of the CodeChecker viewer "
-                                   "server to connect to.")
-
-    common_group.add_argument('-p', '--port',
-                              type=int,
-                              dest="port",
-                              default=8001,
-                              required=False,
-                              help="The port the server is running on.")
+    if needs_product_url is None:
+        # Explicitly not add anything, this boolean acts like a three-way bool.
+        pass
+    elif needs_product_url:
+        # Command connects to a product on a server.
+        common_group.add_argument('--url',
+                                  type=str,
+                                  metavar='PRODUCT_URL',
+                                  dest="product_url",
+                                  default="localhost:8001/Default",
+                                  required=False,
+                                  help="The URL of the product to store the "
+                                       "results for, in the format of "
+                                       "host:port/ProductName.")
+    elif not needs_product_url:
```

Comment:
The comment says above, that `needs_product_url` handles `None`, `True` and `False` separately.

`None` indicates that the command does not connect to any sort of server, and no arguments related to connection are, thus, added.

If it is not `None`, then:
 - `True` means command connects to a product (most of the `cmd` subcommands), and this code includes `--url` to the arg list.
 - `False` means command connects to a server (such as `products`, and `login`), and the code includes `--host` and `--port`.